OnlineStore/OnlineStore.BusinessLogic/Interfaces/ICategoryService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OnlineStore.DTO.DTO;
+
+namespace OnlineStore.BusinessLogic.Interfaces
+{
+    public interface ICategoryService
+    {
+        public Task<CategoryDto?> GetCategoryByIdAsync(Int32 categoryId);
```

Comment:
For asynchronous methods, pass CancellationToken and use it. You can pass it from the controller and propagate it further down the call hierarchy